frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -73,6 +99,15 @@ function ReaderPaging:registerKeyEvents()
             args = -1,
         }
     end
+    --[[ upcoming
+    if Device:hasFiveWay() and not Device:hasKeyboard() then
+        -- targets exclusively kindle 4
+        self.key_events.PrevDocument = {
+            { "ScreenKB", "Back" },
+            event = "PrevDocument",
+            args = 0,
+        }
+    end ]]
```

Comment:
Presumably this is meant to be removed here before merging?

Reply:
well I am having a problem with this one, I can't figure out why but sometimes after you move to the previous document the ScreenKB acts as if it was still being pressed (even though is it physically not), so if you press 'Up' it triggers the shortcut (KB + Up) instead of just the action content selection behind 'Up'. And you can only get out of it by pressing the KB key again. could it be due to the args?

Reply:
Can you see where is the problem lies?

<details>
  <summary>Logs:</summary>

```
05/16/24-10:46:43 DEBUG key event => code: 194 (Press), value: 0, time: 1715856402.888559 
05/16/24-10:46:44 DEBUG key event => code: 194 (Press), value: 1, time: 1715856404.260663 
05/16/24-10:46:44 DEBUG dict lookup word: sardines {
  "189x52+257+379"
} --[[table: 0x43bbac50]] 
05/16/24-10:46:44 DEBUG dict stripped word: sardines 
05/16/24-10:46:44 DEBUG UIManager:preventStandby, counter increased to 1 
05/16/24-10:46:44 DEBUG ImageWidget: _render'ing resources/icons/mdlight/notice-info.svg 40 40 
05/16/24-10:46:44 DEBUG ImageWidget: initial offsets 0 0 
05/16/24-10:46:44 DEBUG show widget: table: 0x43c469b0 
05/16/24-10:46:44 DEBUG setDirty nil from widget table: 0x43c469b0 w/ NO region; dithering: nil 
05/16/24-10:46:44 DEBUG _refresh: Enqueued ui update for region 336.5 390 30 30 dithering: false 
05/16/24-10:46:44 DEBUG setDirty ui from widget ReaderUI w/ region 336.5 390 30 30 dithering: nil 
05/16/24-10:46:44 DEBUG painting widget: ReaderUI 
05/16/24-10:46:44 DEBUG readerview painting "600x800+0+0" to 0 0 
05/16/24-10:46:44 DEBUG CreDocument: goto page 534 flow 0 
No magic header(StarDict's dict ifo file) in ifo file
g_mkdir failed: No such file or directory
05/16/24-10:46:44 DEBUG blitFrom 283 788 0 0 5 5 
05/16/24-10:46:44 DEBUG painting widget: table: 0x43c469b0 
05/16/24-10:46:44 DEBUG triggering refresh {
  mode = "ui",
  region = "30x30+336.5+390"
} --[[table: 0x43c46b60]] 
05/16/24-10:46:44 DEBUG UIManager:_standbyTransition -> PreventStandby 
05/16/24-10:46:44 DEBUG key event => code: 194 (Press), value: 0, time: 1715856404.379884 
05/16/24-10:46:44 DEBUG setDirty via a func from widget table: 0x43c469b0 
05/16/24-10:46:44 DEBUG painting widget: table: 0x43c469b0 
05/16/24-10:46:44 DEBUG blitFrom 75 380 0 0 40 40 
05/16/24-10:46:44 DEBUG _refresh: Enqueued ui update for region 68 361 464 78 dithering: false 
05/16/24-10:46:44 DEBUG triggering refresh {
  mode = "ui",
  region = "464x78+68+361"
} --[[table: 0x43bc5c60]] 
No magic header(StarDict's dict ifo file) in ifo file
05/16/24-10:46:45 DEBUG showing quick lookup window 1 : sardines {
  {
    definition = "\n/sɑɹˈdin/, /sɑːˈdiːn/ \n<ol><li>Any one of several species of small herring which are commonly preserved in olive oil or in tins for food, especially the pilchard, or European sardine <i>Sardina pilchardus</i> (syn. <i>Clupea pilchardus</i>). The California sardine <i>Sardinops sagax</i> (syn. <i>Clupea sagax</i>) is similar. The American sardines of the Atlantic coast are mostly the young of the Atlantic herring and of the menhaden.</li><li><i>(obsolete)</i> carnelian</li><li><i>(figurative)</i> Someone packed or crammed into a small space.</li><li>to fish for sardines</li><li>to pack or cram together tightly.</li></ol></html>",
    dict = "Wiktionary EN-EN",
    is_html = true,
    word = "sardine"
  } --[[table: 0x43bcf660]],
  {
    definition = "\n<ol><li><i>(uncountable)</i> A children's game in which the players hide together, lying side by side.</li></ol></html>",
    dict = "Wiktionary EN-EN",
    is_html = true,
    word = "sardines"
  } --[[table: 0x43bcf748]]
} --[[table: 0x43bcf7f8]] 
05/16/24-10:46:45 DEBUG ImageWidget: _render'ing resources/icons/mdlight/appbar.menu.svg 24 24 
05/16/24-10:46:45 DEBUG renderSVG resources/icons/mdlight/appbar.menu.svg 0.5 48 48 > 24 24 0 0 
05/16/24-10:46:45 DEBUG cache image|resources/icons/mdlight/appbar.menu.svg|24|24false 
05/16/24-10:46:45 DEBUG ImageWidget: initial offsets 0 0 
05/16/24-10:46:45 DEBUG ImageWidget: _render'ing resources/icons/mdlight/close.svg 24 24 
05/16/24-10:46:45 DEBUG ImageWidget: initial offsets 0 0 
05/16/24-10:46:45 DEBUG ImageWidget: _render'ing resources/icons/mdlight/edit.svg 32 32 
05/16/24-10:46:45 DEBUG renderSVG resources/icons/mdlight/edit.svg 0.88888888888889 36 36 > 32 32 0 0 
05/16/24-10:46:45 DEBUG cache image|resources/icons/mdlight/edit.svg|32|32false 
05/16/24-10:46:45 DEBUG ImageWidget: initial offsets 0 0 
05/16/24-10:46:45 DEBUG FocusManager: Move focus position to: 1 , 1 
05/16/24-10:46:45 DEBUG close widget: table: 0x43c469b0 
05/16/24-10:46:45 DEBUG setDirty via a func from widget nil 
05/16/24-10:46:45 DEBUG Lower widget ReaderUI covers the full screen 
05/16/24-10:46:45 DEBUG setDirty nil from widget ReaderUI w/ NO region; dithering: nil 
05/16/24-10:46:45 DEBUG show widget: table: 0x42577ab0 
05/16/24-10:46:45 DEBUG setDirty nil from widget table: 0x42577ab0 w/ NO region; dithering: nil 
05/16/24-10:46:45 DEBUG setDirty via a func from widget table: 0x42577ab0 
05/16/24-10:46:45 DEBUG UIManager:allowStandby, counter decreased to 0 
05/16/24-10:46:45 DEBUG painting widget: ReaderUI 
05/16/24-10:46:45 DEBUG readerview painting "600x800+0+0" to 0 0 
05/16/24-10:46:45 DEBUG CreDocument: goto page 534 flow 0 
05/16/24-10:46:45 DEBUG blitFrom 283 788 0 0 5 5 
05/16/24-10:46:45 DEBUG painting widget: table: 0x42577ab0 
05/16/24-10:46:45 DEBUG blitFrom 53 146 0 0 24 24 
05/16/24-10:46:45 DEBUG blitFrom 523 146 0 0 24 24 
05/16/24-10:46:45 DEBUG blitFrom 516 191 0 0 32 32 
05/16/24-10:46:45 DEBUG Found font: NotoSansArabicUI-Bold.ttf in ./fonts/noto/NotoSansArabicUI-Bold.ttf 
05/16/24-10:46:45 DEBUG _refresh: Enqueued ui update for region 68 361 464 78 dithering: false 
05/16/24-10:46:45 DEBUG update_mode: Update refresh mode ui to flashui 
05/16/24-10:46:45 DEBUG _refresh: Enqueued flashui update for region 40 133 520 509 dithering: false 
05/16/24-10:46:45 DEBUG triggering refresh {
  mode = "flashui",
  region = "520x509+40+133"
} --[[table: 0x42957528]] 
05/16/24-10:46:46 DEBUG UIManager:_standbyTransition -> AllowStandby 
05/16/24-10:46:49 DEBUG input event => type: 4 (EV_MSC), code: 4 (nil), value: 7, time: 1715856409.550987 
05/16/24-10:46:49 DEBUG key event => code: 158 (Back), value: 1, time: 1715856409.551065 
05/16/24-10:46:49 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1715856409.551714 
05/16/24-10:46:49 DEBUG close widget: table: 0x42577ab0 
05/16/24-10:46:49 DEBUG setDirty via a func from widget nil 
05/16/24-10:46:49 DEBUG Lower widget ReaderUI covers the full screen 
05/16/24-10:46:49 DEBUG setDirty nil from widget ReaderUI w/ NO region; dithering: nil 
05/16/24-10:46:49 DEBUG painting widget: ReaderUI 
05/16/24-10:46:49 DEBUG readerview painting "600x800+0+0" to 0 0 
05/16/24-10:46:49 DEBUG CreDocument: goto page 534 flow 0 
05/16/24-10:46:49 DEBUG blitFrom 283 788 0 0 5 5 
05/16/24-10:46:49 DEBUG _refresh: Enqueued flashui update for region 0 0 600 800 dithering: false 
05/16/24-10:46:49 DEBUG triggering refresh {
  mode = "flashui",
  region = "600x800+0+0"
} --[[table: 0x42b5bad0]] 
05/16/24-10:46:49 DEBUG input event => type: 4 (EV_MSC), code: 4 (nil), value: 7, time: 1715856409.760114 
05/16/24-10:46:49 DEBUG key event => code: 158 (Back), value: 0, time: 1715856409.760129 
05/16/24-10:46:49 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1715856409.760131 
05/16/24-10:46:50 DEBUG clear selection 
05/16/24-10:46:50 DEBUG _refresh: Enqueued ui update for region 0 0 600 800 dithering: false 
05/16/24-10:46:50 DEBUG setDirty ui from widget ReaderUI w/ NO region; dithering: nil 
05/16/24-10:46:50 DEBUG painting widget: ReaderUI 
05/16/24-10:46:50 DEBUG readerview painting "600x800+0+0" to 0 0 
05/16/24-10:46:50 DEBUG CreDocument: goto page 534 flow 0 
05/16/24-10:46:50 DEBUG blitFrom 283 788 0 0 5 5 
05/16/24-10:46:50 DEBUG triggering refresh {
  mode = "ui",
  region = "600x800+0+0"
} --[[table: 0x43cd0340]] 
05/16/24-10:46:58 DEBUG input event => type: 4 (EV_MSC), code: 4 (nil), value: 7, time: 1715856418.310986 
05/16/24-10:46:58 DEBUG key event => code: 158 (Back), value: 1, time: 1715856418.311062 
05/16/24-10:46:58 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1715856418.311806 
05/16/24-10:46:58 DEBUG ImageWidget: _render'ing resources/icons/mdlight/notice-question.svg 40 40 
05/16/24-10:46:58 DEBUG renderSVG resources/icons/mdlight/notice-question.svg 0.83333333333333 48 48 > 40 40 0 0 
05/16/24-10:46:58 DEBUG cache image|resources/icons/mdlight/notice-question.svg|40|40true 
05/16/24-10:46:58 DEBUG ImageWidget: initial offsets 0 0 
05/16/24-10:46:58 DEBUG FocusManager: Move focus position to: 1 , 1 
05/16/24-10:46:58 DEBUG show widget: table: 0x43cccfa0 
05/16/24-10:46:58 DEBUG setDirty nil from widget table: 0x43cccfa0 w/ NO region; dithering: nil 
05/16/24-10:46:58 DEBUG setDirty via a func from widget table: 0x43cccfa0 
05/16/24-10:46:58 DEBUG painting widget: table: 0x43cccfa0 
05/16/24-10:46:58 DEBUG blitFrom 75 355 0 0 40 40 
05/16/24-10:46:58 DEBUG _refresh: Enqueued ui update for region 63 343 474 114 dithering: false 
05/16/24-10:46:58 DEBUG triggering refresh {
  mode = "ui",
  region = "474x114+63+343"
} --[[table: 0x42ac1498]] 
05/16/24-10:46:58 DEBUG input event => type: 4 (EV_MSC), code: 4 (nil), value: 7, time: 1715856418.480097 
05/16/24-10:46:58 DEBUG key event => code: 158 (Back), value: 0, time: 1715856418.480112 
05/16/24-10:46:58 DEBUG input event => type: 0 (EV_SYN), code: 0 (SYN_REPORT), value: 0, time: 1715856418.480115 
05/16/24-10:46:59 DEBUG key event => code: 106 (Right), value: 1, time: 1715856419.065170 
05/16/24-10:46:59 DEBUG FocusManager cursor position is: 2 , 1 
05/16/24-10:46:59 DEBUG _refresh: Enqueued fast update for region 0 0 600 800 dithering: false 
05/16/24-10:46:59 DEBUG setDirty fast from widget table: 0x43cccfa0 w/ NO region; dithering: nil 
05/16/24-10:46:59 DEBUG painting widget: table: 0x43cccfa0 
05/16/24-10:46:59 DEBUG blitFrom 75 355 0 0 40 40 
05/16/24-10:46:59 DEBUG triggering refresh {
  mode = "fast",
  region = "600x800+0+0"
} --[[table: 0x43cd5b58]] 
05/16/24-10:46:59 DEBUG key event => code: 106 (Right), value: 0, time: 1715856419.232327 
05/16/24-10:46:59 DEBUG key event => code: 194 (Press), value: 1, time: 1715856419.896948 
05/16/24-10:46:59 DEBUG FocusManager: Send tap to 412 , 428 
05/16/24-10:46:59 DEBUG clear selection 
05/16/24-10:46:59 DEBUG closing reader 
05/16/24-10:46:59 DEBUG closing document 
05/16/24-10:46:59 DEBUG ReaderFooter: unschedule autoRefreshFooter 
05/16/24-10:46:59 DEBUG ReaderCoptListener.headerRefresh unscheduled 
05/16/24-10:46:59 DEBUG cre cache used: ./cache/cr3cache/Black_Snow_--ov_151_.epub.4fc1d9e2.1.cr3 
05/16/24-10:47:00 DEBUG close widget: ReaderUI 
05/16/24-10:47:00 DEBUG DocSettings: Writing to /mnt/us/ePubs/Black Snow - Mikhail Bulgakov (151).sdr/metadata.epub.lua 
05/16/24-10:47:00 DEBUG Tearing down ReaderUI table: 0x40685f98 
05/16/24-10:47:00 DEBUG setDirty nil from widget table: 0x43cccfa0 w/ NO region; dithering: nil 
05/16/24-10:47:00 DEBUG _refresh: Enqueued full update for region 0 0 600 800 dithering: false 
05/16/24-10:47:00 DEBUG close widget: table: 0x43cccfa0 
05/16/24-10:47:00 DEBUG setDirty via a func from widget nil 
05/16/24-10:47:00 INFO  UIManager: No dialogs left to show 
05/16/24-10:47:00 INFO  Tearing down UIManager with exit code: 0 
[ko-input] Closed input device with fd: 6 @ idx: 2
[ko-input] Closed input device with fd: 5 @ idx: 1
[ko-input] Closed input device with fd: 4 @ idx: 0
lipc-wait-event exited normally with status: 0
```

</details>

Reply:
I don't know otoh (am on my phone) but I assume you simply copied it verbatim from one of the shift+ key shortcuts?

Reply:
I have tried different values and they all have the same issue. I thought I understood what the `args` was but apparently not.

Reply:
There's no logging in the modifier codepath, so you might wan to throw some more logging in device/key to figure that one out.

`args` is what gets passed to the event handler.

Reply:
> There's no logging in the modifier codepaths, so you might wan to throw some more logging in device/key to figure that one out.

how do i do that?

Reply:
Either print(...) or logger.dbg()